### PR TITLE
nixUnstable: patch #4470 (ETag)

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Make nixUnstable usable.

https://github.com/NixOS/nix/issues/4469

Proper update seems blocked https://github.com/NixOS/nixpkgs/pull/109309#issuecomment-768331750

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Mic92 — APPROVED** [review]

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -226,6 +226,12 @@ in rec {
         url = "https://github.com/NixOS/nix/commit/d4870462f8f539adeaa6dca476aff6f1f31e1981.patch";
         sha256 = "mTvLvuxb2QVybRDgntKMq+b6da/s3YgM/ll2rWBeY/Y=";
       })
+      # Fix the ETag bug. PR merged. Remove when updating to >= 20210125
+      # https://github.com/NixOS/nixpkgs/pull/109309#issuecomment-768331750
+      (fetchpatch {
+        url = "https://patch-diff.githubusercontent.com/raw/NixOS/nix/pull/4470.diff";
```
Comment (Mic92) [inline pkgs/tools/package-management/nix/default.nix:232]:
```suggestion
        url = "https://github.com/NixOS/nix/commit/c5b42c5a42138329c6d02da0d8a53cb59c6077f4.patch";
```

Comment (Mic92) [reply pkgs/tools/package-management/nix/default.nix:232]: Pull requests in theory could change. This is the merge commit which is immutable.

Comment (Ma27) [reply pkgs/tools/package-management/nix/default.nix:232]: Even though the PR is closed?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111005` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>nix-update</li>
<li>nixFlakes (nixUnstable)</li>
<li>nixpkgs-review</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111005` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>hydra-unstable</li>
<li>nix-update</li>
<li>nixFlakes (nixUnstable)</li>
<li>nixpkgs-review</li>
</ul>
</details>

File: pkgs/tools/package-management/nix/default.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/tools/package-management/nix/default.nix]: Not sure, but better safe than sorry.

**Comment by Mic92** [comment]
backport: https://github.com/NixOS/nixpkgs/pull/111035

**Pull request merged**